python/paddle/fluid/tests/unittests/mkldnn/test_elementwise_add_mkldnn_op.py

```diff
@@ -15,121 +15,318 @@
     def init_input_output(self):
         self.x = np.random.rand(2, 100, 3, 4).astype(self.dtype)
         self.y = np.random.rand(100).astype(self.dtype)
         self.out = self.x + self.y.reshape(1, 100, 1, 1)
 
-    def init_kernel_type(self):
-        self.use_mkldnn = True
+    def init_axis(self):
+        self.axis = 1
 
 
-class TestMKLDNNElementwiseAddOp_broadcast_2(TestElementwiseAddOp_broadcast_2):
+class TestMKLDNNElementwiseAddOp_broadcast_2(TestMKLDNNElementwiseAddOp):
+    def print_info(self):
+        print("### TestMKLDNNElementwiseAddOp_broadcast2 ###")
+
     def init_input_output(self):
-        self.x = np.random.rand(2, 2, 3, 100).astype(self.dtype)
+        self.x = np.random.rand(2, 3, 4, 100).astype(self.dtype)
         self.y = np.random.rand(100).astype(self.dtype)
         self.out = self.x + self.y.reshape(1, 1, 1, 100)
 
-    def init_kernel_type(self):
-        self.use_mkldnn = True
 
+class TestMKLDNNElementwiseAddOp_broadcast_3(TestMKLDNNElementwiseAddOp):
+    def print_info(self):
+        print("### TestMKLDNNElementwiseAddOp_bradcast_3 ###")
 
-class TestMKLDNNElementwiseAddOp_broadcast_3(TestElementwiseAddOp_broadcast_3):
-    def init_kernel_type(self):
-        self.use_mkldnn = True
+    def init_input_output(self):
+        self.x = np.random.rand(2, 30, 40, 5).astype(self.dtype)
+        self.y = np.random.rand(30, 40).astype(self.dtype)
+        self.out = self.x + self.y.reshape(1, 30, 40, 1)
 
+    def init_axis(self):
+        self.axis = 1
 
-class TestMKLDNNElementwiseAddOp_broadcast_4(TestElementwiseAddOp_broadcast_4):
-    def init_kernel_type(self):
-        self.use_mkldnn = True
 
+class TestMKLDNNElementwiseAddOp_broadcast_4(TestMKLDNNElementwiseAddOp):
+    def print_info(self):
+        print("### TestMKLDNNElementwiseAddOp_broadcast_4 ###")
+
+    def init_input_output(self):
+        self.x = np.random.rand(100, 2, 3, 4).astype(self.dtype)
+        self.y = np.random.rand(100, 1).astype(self.dtype)
+        self.out = self.x + self.y.reshape(100, 1, 1, 1)
+
+    def init_axis(self):
+        self.axis = 0
+
+
+class TestMKLDNNElementwiseAddOp_rowwise_add_0(TestMKLDNNElementwiseAddOp):
+    def print_info(self):
+        print("### TestMKLDNNElementwiseAddOp_rowwise_add_0 ###")
 
-class TestMKLDNNElementwiseAddOp_rowwise_add_0(
-        TestElementwiseAddOp_rowwise_add_0):
     def init_input_output(self):
         self.x = np.random.rand(2, 10, 12, 3).astype(self.dtype)
         self.y = np.random.rand(10, 12).astype(self.dtype)
         self.out = self.x + self.y.reshape(1, 10, 12, 1)
 
-    def init_kernel_type(self):
-        self.use_mkldnn = True
+    def init_axis(self):
+        self.axis = 1
 
 
-class TestMKLDNNElementwiseAddOp_rowwise_add_1(
-        TestElementwiseAddOp_rowwise_add_1):
-    def init_kernel_type(self):
-        self.use_mkldnn = True
+@skip_check_grad_ci(
+    reason="[skip shape check] Use y_shape(1) to test broadcast.")
```

Comment:
I'm not sure what is meant here by "Use y_shape(1) to test broadcast".

Reply:
@zhupengyang Could you make the reason more readable?